spec/tapioca/compilers/dsl/mixed_in_class_attributes_spec.rb

```diff
@@ -0,0 +1,130 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "spec_helper"
+
+class Tapioca::Compilers::Dsl::MixedInClassAttributesSpec < DslSpec
+  before do
+    require "active_support/core_ext/class/attribute"
+    require "active_support/concern"
+  end
+
+  describe("#gather_constants") do
+    it("gathers modules that respond to class_attribute") do
+      add_ruby_file("file.rb", <<~RUBY)
+        module ManualIncluded
+          def self.included(base); end
+        end
+
+        module Concern
+          extend ActiveSupport::Concern
+        end
+
+        module SomeOtherModule
+        end
+      RUBY
+
+      assert_includes(gathered_constants, "ManualIncluded")
+      assert_includes(gathered_constants, "Concern")
+      refute_includes(gathered_constants, "SomeOtherModule")
+    end
+
+    it("gathers modules with private included hooks") do
+      add_ruby_file("file.rb", <<~RUBY)
+        module PrivateIncluded
+          def self.included(base); end
+          private_class_method :included
+        end
+      RUBY
+
+      assert_includes(gathered_constants, "PrivateIncluded")
```

Comment:
Same comment as above:
```suggestion
      assert_equal(["PrivateIncluded"], gathered_constants)
```